caterva2/services/sub.py

```diff
@@ -398,13 +399,18 @@ async def get_roots(user: db.User = Depends(current_active_user)) -> dict:
     dict
         The dict of roots.
     """
-    if not user:
-        return database.roots
+    # For some reason, '@public' could not be found in the database
+    # (even though it is injected manually in main).  I suppose that means
+    # I don't completely understand the different ways to register roots.
+    if '@public' not in database.roots:
+        root = models.Root(name='@public', http='', subscribed=True)
+        database.roots[root.name] = root
```

Comment:
database.roots are the roots from publishers; while personal, shared and public are hardcoded, they should not be included in database.roots, that's why we have a copy() line just below, so personal and shared are "injected" wihtout modifying database.roots, public should be the same

This explains my earlier comment about the template, public should not database.roots

Reply:
Ditto